x/evm/keeper/msg_server.go

```diff
@@ -1052,3 +1052,38 @@ func (s msgServer) AddChain(c context.Context, req *types.AddChainRequest) (*typ
 
 	return &types.AddChainResponse{}, nil
 }
+
+func (s msgServer) RetryFailedEvent(c context.Context, req *types.RetryFailedEventRequest) (*types.RetryFailedEventResponse, error) {
+	ctx := sdk.UnwrapSDKContext(c)
+
+	chain, ok := s.nexus.GetChain(ctx, req.Chain)
+	if !ok {
+		return nil, fmt.Errorf("%s is not a registered chain", req.Chain)
+	}
+
+	if err := validateChainActivated(ctx, s.nexus, chain); err != nil {
+		return nil, err
+	}
+
+	keeper := s.ForChain(chain.Name)
+
+	event, ok := keeper.GetEvent(ctx, req.EventID)
+	if !ok {
+		return nil, fmt.Errorf("event %s not found for chain %s", req.EventID, req.Chain)
+	}
+
+	if event.Status != types.EventFailed {
+		return nil, fmt.Errorf("event %s is not a failed event", req.EventID)
+	}
+
+	event.Status = types.EventConfirmed
+	keeper.GetConfirmedEventQueue(ctx).Enqueue(getEventKey(req.EventID), &event)
+
+	s.Logger(ctx).Info(
+		fmt.Sprintf("retry failed event"),
```

Comment:
```suggestion
		fmt.Sprintf("re-queued failed event"),
```
so that it's clearer that the handling is async